doc/guides/west/manifest.rst

```diff
@@ -181,55 +201,87 @@ In this manifest:
-- ``west-commands``: Optional. If given, a relative path to a YAML file within
-  the project which describes additional west commands provided by that
-  project. This file is named :file:`west-commands.yml` by convention. See
-  :ref:`west-extensions` for details.
-- ``import``: Optional. If ``true``, imports projects from manifest files in
-  the given repository into the current manifest. See
-  :ref:`west-manifest-import` for details.
-- ``groups``: Optional, a list of groups the project belongs to. See
-  :ref:`west-manifest-groups` for details.
-- ``submodules``: Optional. You can use this to make ``west update`` also
-  update `Git submodules`_ defined by the project. See
-  :ref:`west-manifest-submodules` for details.
+The available project keys and their usage are in the following table.
+Sometimes we'll refer to the ``defaults`` subsection; it will be described
+next.
+
+.. list-table:: projects elements keys
+   :header-rows: 1
+   :widths: 1 5
+
+   * - Key(s)
+     - Description
+
+   * - ``name``
+     - Mandatory; a unique name for the project. The name cannot be one of the
+       reserved values "west" or "manifest". The name must be unique in the
+       manifest file.
+
+   * - ``remote``, ``url``
+     - Mandatory (one of the two, but not both).
+
+       If the project has a ``remote``, that remote's ``url-base`` will be
+       combined with the project's ``name`` (or ``repo-path``, if it has one)
+       to form the fetch URL instead.
+
+       If the project has a ``url``, that's the complete fetch URL for the
+       remote Git repository.
+
+       If the project has neither, the ``defaults`` section must specify a
+       ``remote``, which will be used as the the project's remote. Otherwise,
+       the manifest is invalid.
+
+   * - ``repo-path``
+     - Optional. If given, this is concatenated on to the remote's
+       ``url-base`` instead of the project's ``name`` to form its fetch URL.
+       Projects may not have both ``url`` and ``repo-path`` attributes.
+
+   * - ``revision``
+     - Optional. The Git revision that ``west update`` should
+       check out. This will be checked out as a detached HEAD by default, to
+       avoid conflicting with local branch names. If not given, the
+       ``revision`` value from the ``defaults`` subsection will be used if
+       present.
+
+       A project revision can be a branch, tag, or SHA.
+
+       The default ``revision`` is ``master`` if not otherwise specified.
+
+   * - ``path``
+     - Optional. Relative path specifying where to clone the repository
+       locally, relative to the top directory in the west workspace. If missing,
+       the project's ``name`` is used as a directory name.
+
+   * - ``clone-depth``
+     - Optional. If given, a positive integer which creates a shallow history
+       in the cloned repository limited to the given number of commits. This
+       can only be used if the ``revision`` is a branch or tag.
```

Comment:
Curious how this will interact with the incoming `shallow config`...

Reply:
> Curious how this will interact with the incoming `shallow config`...

This is basically a failed experiment that no one uses, so it's a bit of a tree falling in the forst.